decidim-design/app/helpers/decidim/design/share_helper.rb

```diff
@@ -17,15 +17,23 @@ def share_sections
                 type: :cell_table,
                 options: { headings: [t("decidim.design.helpers.share_button")] },
                 cell_snippet: {
-                  cell: "decidim/share_button",
-                  args: [],
-                  call_string: 'cell("decidim/share_button", nil)'
+                  cell: "decidim/share_widget",
+                  args: [shareable_resource],
+                  call_string: 'cell("decidim/share_widget", resource)'
                 }
               }
             ]
           }
         ]
       end
+
+      protected
+
+      def shareable_resource
```

Comment:
This method is the only one that I have doubts to be honest, as in a new empty database this would break.

I have two alternatives here (by order of preference):

1. To just give a dummy class that renders a link to whatever (I don't know if this is compatible with the ShortUrl API)
2. To add an if in the HTML that make these checks and add a text with the edge case, i.e. "We can not show this field as you don't have contents in your database. Create a Debate, Meeting or Proposal to show this widget"